presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -330,7 +330,7 @@ public CounterStat getHighMemorySplitSource()
 
                 if (bucketProperty.isPresent()) {
                     Optional<HiveBucketProperty> partitionBucketProperty = partition.getStorage().getBucketProperty();
-                    if (!partitionBucketProperty.isPresent()) {
+                    if (!partitionBucketProperty.isPresent() && isBucketExecutionEnabled(session)) {
```

Comment:
Do we need this change? If `isBucketExecutionEnabled(session)` is `false`, the `Optional<HiveBucketHandle> bucketHandle` will be set to `empty()` here: https://github.com/prestodb/presto/pull/12480/files#diff-5b2f67483a7afa9203204186749ecf84R185, so the `bucketProperty` will be `empty()` as well (https://github.com/prestodb/presto/pull/12480/files#diff-5b2f67483a7afa9203204186749ecf84R193), so the `bucketProperty.isPresent()` `https://github.com/prestodb/presto/pull/12480/files#diff-5b2f67483a7afa9203204186749ecf84R331` wont pass.